cpp/include/cudf/detail/iterator.cuh

```diff
@@ -153,6 +263,53 @@ auto inline make_validity_iterator(column_device_view const& column)
       validity_accessor{column});
 }
 
+/**
+ * @brief Constructs a constant iterator over a scalar's value.
+ *
+ * Dereferencing the returned iterator returns a `Element`.
+ * 
+ * For `p = *(iter + i)`, `p` is the value stored in the scalar.
+ *
+ * @throws `cudf::logic_error` if scalar datatype and Element type mismatch.
+ * @throws `cudf::logic_error` if scalar is null.
+ *
+ * @tparam Element The type of elements in the scalar
+ * @param scalar_value The scalar to iterate
+ * @return auto Iterator that returns scalar value
+ */
+template <typename Element>
+auto inline make_scalar_iterator(scalar const& scalar_value)
+{
+  CUDF_EXPECTS(data_type(experimental::type_to_id<Element>()) == scalar_value.type(), "the data type mismatch");
+  CUDF_EXPECTS(scalar_value.is_valid(), "the scalar value must be valid");
+  using ScalarType = experimental::scalar_type_t<Element>;
+  return thrust::make_constant_iterator(static_cast<ScalarType const*>(&scalar_value)->value());
+}
+
+/**
+ * @brief Constructs a constant pair iterator over a scalar's value and its validity.
+ *
+ * Dereferencing the returned iterator returns a `thrust::pair<Element, bool>`.
+ *  
+ * If scalar is valid, then for `p = *(iter + i)`, `p.first` contains 
+ * the value of the scalar and `p.second == true`.
+ *
+ * Else, if the scalar is null, then the value of `p.first` is undefined and `p.second == false`. 
+ * 
+ * @throws `cudf::logic_error` if scalar datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the scalar
+ * @param scalar_value The scalar to iterate
+ * @return auto Iterator that returns scalar, and validity of the scalar in a pair
+ */
+template <typename Element>
+auto inline make_scalar_pair_iterator(scalar const& scalar_value) {
```

Comment:
This should just be an overload of `make_pair_iterator` in order to allow transparently using one or the other in a template function, e.g.,
```c++
template <typename ColumnOrScalar>
void foo(ColumnOrScalar x){
  auto iter = make_pair_iterator(x);
}
```
```suggestion
auto inline make_pair_iterator(scalar const& scalar_value) {
```